### PR TITLE
Swift Package Manager support

### LINKED ISSUE #1387: Library Does Not Support Installation Via Swift Package Manager
### Bug Report
Currently, the library does not support installation via Swift Package Manager due to SPM's inability to resolve submodules [see here for more info](https://forums.swift.org/t/package-with-git-submodule-doesnt-build-correctly/27852). We will have to move the submodule code into the library.

##### Reproduction Steps
1. Set up Xcode 11 with a sample app
2. Attempt to install sdl_ios via SPM

##### Expected Behavior
The imported package works correctly

##### Observed Behavior
The imported package fails to work correctly due to failures around the BSON lib.

##### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 6.4-develop
* Testing Against: Xcode 11

### PR DESCRIPTION
**Pull request by joeljfischer**
Fixes #1387 

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
No unit tests changed, but some merge conflicts were fixed

#### Core Tests
A SPM test app was run with basic text update and lock screen tests.

Core version / branch / commit hash / module tested against: Manticore (Core v6.1.1)
HMI name / version / branch / commit hash / module tested against: Manticore (generic_hmi v0.8.1)

### Summary
This PR adds support for Swift Package Manager. Note that the bson_c_lib dependency will need to be updated when [this PR](https://github.com/smartdevicelink/bson_c_lib/pull/49) is merged and release 1.2.2 occurs.

##### Enhancements
* Adds support for using SDL iOS as a Swift Package through Xcode 12+.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
I left one question.

File: SmartDeviceLink-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -300,7 +299,7 @@
 		4A457DD524A3C16E00386CBA /* SDLLifecycleMobileHMIStateHandlerSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = 4A457DD424A3C16E00386CBA /* SDLLifecycleMobileHMIStateHandlerSpec.m */; };
 		4A457DD724A3CCED00386CBA /* SDLLifecycleSystemRequestHandlerSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = 4A457DD624A3CCED00386CBA /* SDLLifecycleSystemRequestHandlerSpec.m */; };
 		4A457DD924A5137100386CBA /* SDLLifecycleProtocolHandlerSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = 4A457DD824A5137100386CBA /* SDLLifecycleProtocolHandlerSpec.m */; };
-		4A877499251BBE2400A8BEBB /* SmartDeviceLink.h in Headers */ = {isa = PBXBuildFile; fileRef = 4A877498251BBE2400A8BEBB /* SmartDeviceLink.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		4A877499251BBE2400A8BEBB /* (null) in Headers */ = {isa = PBXBuildFile; settings = {ATTRIBUTES = (Public, ); }; };
```
[the remark this answers is not in the document]
Comment (joeljfischer) [reply SmartDeviceLink-iOS.xcodeproj/project.pbxproj]: Done

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
SPM should be added to the list of supported dependency managers in the **README** file. If there is a SPM badge it should also be added.

**Comment by joeljfischer** [comment]
Fixed

File: README.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
-[![Accio supported](https://img.shields.io/badge/Accio-supported-0A7CF5.svg?style=flat)](https://github.com/JamitLabs/Accio)
+[![Accio compatible](https://img.shields.io/badge/Accio-compatible-4BC51D.svg?style=flat)](https://github.com/JamitLabs/Accio)
```
Comment (NicoleYarroch) [inline README.md:1]: This paragraph also needs to get updated:

> There are instructions on how to install using the dependency managers CocoaPods, Carthage, and Accio, as well as how to install the library framework manually.

**Review by NicoleYarroch — CHANGES_REQUESTED** [review]
One change requested.

**Review by NicoleYarroch — APPROVED** [review]

**Pull request merged**